File: smash-ranks-android/app/src/test/java/com/garpr/android/BaseTest.java
package com.garpr.android;

import android.app.Application;

import com.garpr.android.dagger.TestAppComponent;
import com.garpr.android.dagger.TestAppComponentHandle;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import androidx.annotation.NonNull;
import androidx.test.core.app.ApplicationProvider;

@Config(application = TestApp.class)
@RunWith(RobolectricTestRunner.class)
public abstract class BaseTest implements TestAppComponentHandle {

    @NonNull
    @Override
    public TestAppComponent getTestAppComponent() {
        final Application application = ApplicationProvider.getApplicationContext();
        return ((TestAppComponentHandle) application).getTestAppComponent();
    }

    @Before
    public void setUp() {
        // intentionally empty
    }

}

File: smash-ranks-android/app/src/test/java/com/garpr/android/BaseTest.kt
package com.garpr.android

import com.garpr.android.koin.configModule
import com.garpr.android.koin.databaseModule
import com.garpr.android.koin.managersModule
import com.garpr.android.koin.miscModule
import com.garpr.android.koin.networkingModule
import com.garpr.android.koin.preferencesModule
import com.garpr.android.koin.repositoriesModule
import com.garpr.android.koin.syncModule
import com.garpr.android.koin.viewModelsModule
import org.junit.Before
import org.koin.core.context.startKoin
import org.koin.test.AutoCloseKoinTest

abstract class BaseTest : AutoCloseKoinTest() {

    private fun initializeKoin() {
        startKoin {
            modules(listOf(configModule, databaseModule, managersModule, miscModule,
                    networkingModule, preferencesModule, repositoriesModule, syncModule,
                    viewModelsModule))
        }
    }

    @Before
    open fun setUp() {
        initializeKoin()
    }

}
